src/util.js

```diff
@@ -1,12 +1,16 @@
-export function _normalizeAbsolutePath(path, hasPushState) {
+export function _normalizeAbsolutePath(path, hasPushState, absolute = false) {
   if (!hasPushState && path[0] !== '#') {
     path = '#' + path;
   }
 
+  if (hasPushState && absolute) {
+    path = path.substring(1, path.length);
+  }
+
   return path;
 }
 
-export function _createRootedPath(fragment, baseUrl, hasPushState) {
+export function _createRootedPath(fragment, baseUrl, hasPushState, absolute) {
```

Comment:
I'm not sure if this is possible. The `hasPushState` argument actually determines whether or not we add the hash character to a generated URL - it's only passed to `_createdRootedPath` to get passed to the normalize function. I'm pretty sure that's necessary, and there's no way that having the absolute root URL would let us know whether or not to use the hash.